lib/matplotlib/axes/_base.py

```diff
@@ -699,6 +699,10 @@ def __init__(self, fig, rect,
                         rcParams['ytick.major.right']),
             which='major')
 
+    def __init_subclass__(cls, **kwargs):
+        cls.__subclass_uses_cla = 'cla' in cls.__dict__
```

Comment:
If true we should issue a pending deprecation so that downstream libraries can start moving to the canonical interface.